src/v/kafka/server/group_manager.cc

```diff
@@ -499,28 +472,17 @@ ss::future<> group_manager::recover_partition(
         for (auto& [id, timeout] : group_stm.timeouts()) {
             group->try_set_timeout(id, timeout);
         }
-    }
 
-    /*
-     * <kafka>if the cache already contains a group which should be removed,
-     * raise an error. Note that it is possible (however unlikely) for a
-     * consumer group to be removed, and then to be used only for offset storage
-     * (i.e. by "simple" consumers)</kafka>
-     */
-    for (auto& [group_id, group_stm] : ctx.groups) {
         if (group_stm.is_removed()) {
-            if (_groups.contains(group_id) && group_stm.offsets().size() > 0) {
+            if (group_stm.offsets().size() > 0) {
                 klog.warn(
-                  "Unexpected active group unload {} loading {}",
+                  "Unexpected active group unload {} while loading {}",
                   group_id,
                   p->partition->ntp());
             }
         }
     }
-
-    _groups.rehash(0);
```

Comment:
nit: Did you mean to omit this intentionally

Reply:
@mmaslankaprv : Oh I noticed this comment after hitting merge.

Reply:
>@mmaslankaprv : Oh I noticed this comment after hitting merge.

it's ok to drop it or add it back. i don't think we've ever worked through if the optimization is useful

Reply:
i dropped this intentionally to prevent going through the whole map once again